### PR TITLE
Auth service

### PR DESCRIPTION
**Pull request by rivernate**

### CODE REVIEW

File: lib/ims/lti/services/authentication_service.rb [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+          sub: @_sub,
+          aud: @_aud.to_s,
+          iat: Time.now.to_i,
+          exp: 1.minute.from_now,
+          jti: SecureRandom.uuid
+        )
+        assertion = assertion.sign(@_secret, :HS256).to_s
+        request = {
+          body: {
+            grant_type: 'urn:ietf:params:oauth:grant-type:jwt-bearer',
```
Comment (westonkd) [inline lib/ims/lti/services/authentication_service.rb]: Do we want to have an option to set this to `authorization_code` or any other type?

File: lib/ims/lti/services/authentication_service.rb [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+          iss: @_iss,
+          sub: @_sub,
+          aud: @_aud.to_s,
+          iat: Time.now.to_i,
+          exp: 1.minute.from_now,
+          jti: SecureRandom.uuid
+        )
+        assertion = assertion.sign(@_secret, :HS256).to_s
+        request = {
+          body: {
```
Comment (westonkd) [inline lib/ims/lti/services/authentication_service.rb]: Should we include a way to send other parameters (i.e. `code`)

Comment (rivernate) [inline lib/ims/lti/services/authentication_service.rb]: done

File: lib/ims/lti/services/authentication_service.rb (hunk 1 above)
Comment (rivernate) [inline lib/ims/lti/services/authentication_service.rb]: done

File: lib/ims/lti/services/authentication_service.rb [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+          sub: sub,
+          aud: aud.to_s,
+          iat: Time.now.to_i,
+          exp: 1.minute.from_now,
+          jti: SecureRandom.uuid
+        )
+        assertion.merge!(@additional_claims)
+        assertion = assertion.sign(@secret, :HS256).to_s
+        body = {
+          grant_type: grant_type,
```
Comment (westonkd) [inline lib/ims/lti/services/authentication_service.rb]: Can we add a way to include arbitrary params sent in the body? We will need a way to send the `code` param when we retrieve an access token before requesting the TCP

**Pull request merged**